server/controllers/userAuth.js

```diff
@@ -365,9 +365,59 @@ const resendRecoveryLink = (req, resp, next) => {
     });
 };
 
+const getUserDetails = (req, resp) => {
+  const { user } = req;
+
+  if (user) {
+    const { activatedAt, createdAt, email, password } = user;
+    const activationDate = activatedAt || createdAt;
+    const isPasswordSet = password !== undefined;
+
+    return resp.send({ activationDate, email, isPasswordSet });
+  }
+
+  return resp.sendStatus(204);
+};
+
+const changePassword = (req, resp) => {
+  const { password, newPassword, newPasswordConfirm } = req.body;
+  const errors = {};
+  const { email } = req.user;
+
+  errors.isNewPasswordError = !validatePasswordStructure(newPassword);
+  errors.isNewConfirmPasswordError = newPassword !== newPasswordConfirm;
+
+  if (_some(errors, error => error)) {
+    return resp.status(406).send(errors);
```

Comment:
404 - Not Acceptable is used for content negotiation. Consider returning plain 400 Bad request error - https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/406